args.go

```diff
@@ -128,7 +128,7 @@ func (a *ArgumentBase[T, C, VC]) Parse(s []string) ([]string, error) {
 
 	if a.Values == nil {
 		a.Values = &values
-	} else {
+	} else if count > 0 {
```

Comment:
shouldnt the if count be outside the if a.Values ...?

Reply:
My thinking was that there's only one out of four cases where we _don't_ want to mutate `a.Values`: if `a.Values` has defaults && no args were supplied to override those. So the second branch of this conditional just needs to check if any args were collected before mutating `a.Values`; otherwise we should leave the defaults unmodified.

I don't feel strongly about how the logic is implemented though, if you think there's a cleaner way to phrase this.

Reply:
Thats fine @joshfrench . Lets leave it as it is